controllers/discovery/initDiscoveryTopicData.js

```diff
@@ -13,6 +13,15 @@ const InitDiscoveryTopicData = async (req, res) => {
   const userId = req.userId;
 
   try {
+    let totalDataQuery = `SELECT 
+                              count(C.topic_id) as total_data
+                          FROM user_topics A 
+                          INNER JOIN user_topics B 
+                              ON A.topic_id = B.topic_id 
+                              AND A.user_id = :userId
+                          RIGHT JOIN topics C 
+                              ON C.topic_id = A.topic_id`;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The SQL query for `totalDataQuery` seems to be incorrect. It's joining `user_topics` with itself and then right joining with `topics`. This could lead to incorrect results. You should directly join `user_topics` with `topics` based on `topic_id` and filter by `userId`.
```diff
-    let totalDataQuery = `SELECT 
-                              count(C.topic_id) as total_data
-                          FROM user_topics A 
-                          INNER JOIN user_topics B 
-                              ON A.topic_id = B.topic_id 
-                              AND A.user_id = :userId
-                          RIGHT JOIN topics C 
-                              ON C.topic_id = A.topic_id`;
+    let totalDataQuery = `SELECT 
+                              count(A.topic_id) as total_data
+                          FROM user_topics A 
+                          INNER JOIN topics B 
+                              ON A.topic_id = B.topic_id 
+                          WHERE A.user_id = :userId`;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
@n-permana for total topics, you don't have to join `user_topics`, just count topics will be enough